src/tiled/wangfiller.cpp

```diff
@@ -285,7 +287,7 @@ WangId WangFiller::wangIdFromSurroundings(const TileLayer &back,
     getSurroundingPoints(point, mStaggeredRenderer, mStaggerAxis, adjacentPoints);
 
     for (int i = 0; i < 8; ++i) {
-        if (!fillRegion.contains(adjacentPoints[i]) && back.contains(adjacentPoints[i]))
+        if (!fillRegion.contains(adjacentPoints[i]) && (mIsInfinite || back.contains(adjacentPoints[i])))
```

Comment:
Since `cellAt` will just return an empty cell and no longer asserts when the position is outside of the layer, the ` && (mIsInfinite || back.contains(adjacentPoints[i]))` part can be left out entirely.